open_spiel/games/2048.cc

```diff
@@ -0,0 +1,383 @@
+  if (!WithinBounds(x, y))
+    return 0;
+  return BoardAt(x, y).value;
+}
+
+void TwoZeroFourEightState::DoApplyAction(Action action) {
+  if (IsChanceNode()) {
+    // The original 2048 game starts with two random tiles
+    if (!extra_chance_turn_) {
+      current_player_ = 0;
+    }
+    extra_chance_turn_ = false;
+    
+    if (action == kNoCellAvailableAction) {
+      return;
+    }
+    ChanceAction chance_action = SpielActionToChanceAction(action);
+    SetBoard(chance_action.row, chance_action.column,
+        Tile(chance_action.is_four ? kChanceTiles[1] : kChanceTiles[0], false));
+    return;
+  }
+  std::vector<std::vector<int>> traversals = BuildTraversals(action);
+  PrepareTiles();
+  for (int x : traversals[0]) {
+    for (int y : traversals[1]) {
+      int tile = GetCellContent(x, y);
+      if (tile > 0) {
+        bool moved = false;
+        std::vector<Coordinate> positions = FindFarthestPosition(x, y, action);
+        Coordinate farthest_pos = positions[0];
+        Coordinate next_pos = positions[1];
+        int next_cell = GetCellContent(next_pos.x, next_pos.y);
+        if (next_cell > 0 && next_cell == tile
+            && !BoardAt(next_pos.x, next_pos.y).is_merged) {
+          int merged = tile * 2;
+          SetBoard(next_pos.x, next_pos.y, Tile(merged, true));
+          moved = true;
+        } else if (farthest_pos.x != x || farthest_pos.y != y){
+          SetBoard(farthest_pos.x, farthest_pos.y, Tile(tile, false));
+          moved = true;
+        }
+        if (moved) {
+          SetBoard(x, y, Tile(0, false));
+          current_player_ = kChancePlayerId;
+        }        
+      }
+    }
+  }  
+}
+
+std::string TwoZeroFourEightState::ActionToString(Player player,
+                                          Action action_id) const {
+  if (IsChanceNode()) {
+    if (action_id == kNoCellAvailableAction) {
+      return "No Cell Available";
+    }
+    ChanceAction chance_action = SpielActionToChanceAction(action_id);
+    return absl::StrCat(std::to_string(chance_action.is_four ? 4 : 2), 
+        " added to row ", std::to_string(chance_action.row + 1),
+        ", column ", std::to_string(chance_action.column + 1));
+  }
+  switch (action_id) {
+    case kMoveUp:
+      return "Up";
+      break;
+    case kMoveRight:
+      return "Right";
+      break;
+    case kMoveDown:
+      return "Down";
+      break;
+    case kMoveLeft:
+      return "Left";
+      break;
+    default:
+      return "Invalid action";
+      break;
+  }  
+}
+
+int TwoZeroFourEightState::AvailableCellCount() const {
+  int count = 0;
+  for (int r = 0; r < kDefaultRows; r++) {
+    for (int c = 0; c < kDefaultColumns; c++) {
+      if (BoardAt(r, c).value == 0) {
+        count++;
+      }
+    }
+  }
+  return count;
+}
+
+ActionsAndProbs TwoZeroFourEightState::ChanceOutcomes() const {
+  ActionsAndProbs action_and_probs;
+  int count = AvailableCellCount();
+  if (count == 0) {
+    action_and_probs.reserve(1);
```

Comment:
this should work and is simpler: `return {{kNoCellAvailableAction, 1.0}};`

Reply:
Have changed this